enterprise/cli/workspaceproxy.go

```diff
@@ -0,0 +1,156 @@
+		formatter             = cliui.NewOutputFormatter(
+			// Text formatter should be human readable.
+			cliui.ChangeFormatterData(cliui.TextFormat(), func(data any) (any, error) {
+				response, ok := data.(codersdk.CreateWorkspaceProxyResponse)
+				if !ok {
+					return nil, xerrors.Errorf("unexpected type %T", data)
+				}
+				return fmt.Sprintf("Workspace Proxy %q registered successfully\nToken: %s", response.Proxy.Name, response.ProxyToken), nil
+			}),
+			cliui.JSONFormat(),
+			// Table formatter expects a slice, make a slice of one.
+			cliui.ChangeFormatterData(cliui.TableFormat([]codersdk.CreateWorkspaceProxyResponse{}, []string{"proxy name", "proxy url", "proxy token"}),
+				func(data any) (any, error) {
+					response, ok := data.(codersdk.CreateWorkspaceProxyResponse)
+					if !ok {
+						return nil, xerrors.Errorf("unexpected type %T", data)
+					}
+					return []codersdk.CreateWorkspaceProxyResponse{response}, nil
+				}),
+		)
+	)
+
+	client := new(codersdk.Client)
+	cmd := &clibase.Cmd{
+		Use:   "create",
+		Short: "Create a workspace proxy",
+		Middleware: clibase.Chain(
+			clibase.RequireNArgs(0),
+			r.InitClient(client),
+		),
+		Handler: func(inv *clibase.Invocation) error {
+			ctx := inv.Context()
+			resp, err := client.CreateWorkspaceProxy(ctx, codersdk.CreateWorkspaceProxyRequest{
+				Name:             proxyName,
+				DisplayName:      displayName,
+				Icon:             proxyIcon,
+				URL:              proxyURL,
+				WildcardHostname: proxyWildcardHostname,
+			})
+			if err != nil {
+				return xerrors.Errorf("create workspace proxy: %w", err)
+			}
+
+			var output string
+			if onlyToken {
+				output = resp.ProxyToken
+			} else {
+				output, err = formatter.Format(ctx, resp)
+				if err != nil {
+					return err
+				}
+			}
+
+			_, err = fmt.Fprintln(inv.Stdout, output)
+			return err
+		},
+	}
+
+	formatter.AttachOptions(&cmd.Options)
+	cmd.Options.Add(
+		clibase.Option{
+			Flag:        "name",
+			Description: "Name of the proxy. This is used to identify the proxy.",
+			Value:       clibase.StringOf(&proxyName),
+		},
+		clibase.Option{
+			Flag:        "display-name",
+			Description: "Display of the proxy. If omitted, the name is reused as the display name.",
+			Value:       clibase.StringOf(&displayName),
+		},
+		clibase.Option{
+			Flag:        "icon",
+			Description: "Display icon of the proxy.",
+			Value:       clibase.StringOf(&proxyIcon),
+		},
+		clibase.Option{
+			Flag:        "access-url",
+			Description: "Access URL of the proxy.",
+			Value:       clibase.StringOf(&proxyURL),
+		},
+		clibase.Option{
+			Flag:        "wildcard-access-url",
+			Description: "(Optional) Access url of the proxy for subdomain based apps.",
+			Value:       clibase.StringOf(&proxyWildcardHostname),
+		},
+		clibase.Option{
+			Flag:        "only-token",
+			Description: "Only print the token. This is useful for scripting.",
+			Value:       clibase.BoolOf(&onlyToken),
+		},
```

Comment:
Honestly seems unnecessary given you support JSON output.

Reply:
This is so easy:
```
proxy_session_token=$(coder proxy create --only-token)
```

Otherwise I need to chain with jq. This is just nice imo.